app/models/agents/website_agent.rb

```diff
@@ -106,24 +106,41 @@ class WebsiteAgent < Agent
 
       Set `http_success_codes` to an array of status codes (e.g., `[404, 422]`) to treat HTTP response codes beyond 200 as successes.
 
+      The `interpolate` option is the way to format extracted data, and add meta information to each resulted payload.  Its value must be a hash, whose key-value pairs are interpolated after extraction and added to each event payload.  e.g.:
+
+          "interpolate": {
```

Comment:
One big problem in decoupling fetching from parsing is that you have to save and carry the full content of a web page to the parser via event for each check, regardless of whether there has been a change in the results of scraping.  It can be a big waste if you check many pages for changes frequently.  For that reason, WebsiteAgent has `on_change` mode that prevents it from wasting storage.  If you are watching a product page for a price change, there's no point in saving every snapshot of the whole page when the price hasn't changed since the last check, so I tell WebsiteAgent where to focus by a CSS selector and not to care about anything else, and it only emits an event if it finds an update in the price.  This is what I think is nice about fetching and parsing done in one place.

So, if we really want to breaking up WebsiteAgent into the fetcher and the parser, we may need to introduce short life events to minimize waste of storage.  Perhaps there could be a "propagate and delete immediately" option, which would make an agent instantiate each event only on memory without saving in the database and invoke the receiver agents passing the payload, but it is a casual idea just popped into my mind, needs more insight and consideration.